### ISSUE TITLE
Nested forms validate uniqueness based on database before transaction

### ISSUE THREAD

**Issue by onebree** [ryanb/nested_form#356]
**Rails 4.2.0, nested_form 0.3.2**

**TL;DR** - The `validates_uniqueness_on` method only compares current records at their current state _before_ the form was submitted.

```
Location has_many :dids
Location has_many :screened_dids
Location accepts_nested_attributes_for :screened_dids

Did has_one :screened_did

ScreenedDid belongs_to :location
ScreenedDid belongs_to :did
ScreenedDid validates_uniqueness_of :did_id
```

Given the above associations, say a Location has 5 DIDs and 0 Screened DIDs. On the nested form, I add 5 rows of Screened DIDs, all having the same selected `:did_id`. Because there were zero Screened DID records for this location before the transaction, all nested objects will be created with different ids but the same `:did_id`. 

If I have 1 Screened DID present with a `:did_id` of `500`, and I attempt to create/update _one_ other object with the same, the validation will kick in.

If I have 2 Screened DID objects with different `:did_id` (`500` and `501`), and I update both to have the same (`502`), then the validation will fail, like in the first scenario.
